src/ucs/sys/topo/base/topo.c

```diff
@@ -306,6 +372,65 @@ ucs_topo_get_distance_sysfs(ucs_sys_device_t device1,
     return UCS_OK;
 }
 
+ucs_numa_distance_t
+ucs_topo_numa_distance(ucs_numa_node_t source, ucs_numa_node_t dest)
+{
+    ucs_numa_distance_t distance;
+
+    ucs_spin_lock(&ucs_topo_global_ctx.lock);
+    distance = ucs_numa_distance(source, dest);
+    ucs_spin_unlock(&ucs_topo_global_ctx.lock);
+
```

Comment:
it's weird that hash table is defined in numa module and it's not thread safe, and it's protected by a lock in another module
need to move the lock also to numa, so numa layer will be thread-safe API
so probably don't even need this wrapper function (ucs_topo_numa_distance)

Reply:
It was in numa.c, but it required spinlock, init and cleanup functions. 
We moved it to topo.c to get rid of the spinlock and the init/cleanup functions.
The downside is that it's a bit weird like you said, and also the implementation is now part of the exposed ucs h files.